Feat: usa collection novidades

ThaisMap opened the pull request:
- Adiciona service para consultar se tem alguma novidade para exibir a partir da data atual, página informada e novidades dispensadas pelo usuário
- Ajusta componente novidade para exibir dados da collection ao invés de dados fixos
- Ajusta páginas do app para exibir o componente Novidade

ThaisMap committed:
fix: corrige alguns typos

ThaisMap committed:
feat: adiciona service getNovidadeValida

- filtra novidade pelo path, data e se foi dispensada pelo usuário

- para realizar a consulta composta foi necessário criar um index no firebase

ThaisMap committed:
feat: usa collection no componente Novidade

- ajusta componente para exibir dados da collection
- ajusta páginas para exibir o componente Novidade

ThaisMap committed:
fix: ajusta verificação de autoDispensar

- faz a verificação quando estiver saindo da página

ThaisMap commented:
Ref #178378564

kellynvd commented on src/services/novidades/GetNovidadeValida.ts:
```diff
@@ -0,0 +1,30 @@
+  INovidadesRepository
+} from 'src/repositories/NovidadesRepository'
+
+interface IGetNovidadesValida {
```
Como é śo uma novidade acho que aqui pode ser Novidade no singular, e o  retorno também: `return novidadeValida`

ThaisMap replied:
Ok

kellynvd reviewed (changes requested):
Fiz alguns testes e anotei alguns pontos:

- [x] Criei uma novidade pra `/configuracoes` mas não apareceu
- [x] Criei duas novidades marcando a opção "Aparece só uma vez", mas elas continuam aparecendo mesmo saindo e voltando da página
- [x] Criei uma novidade com data final de hoje mas ela não aparece
    - Talvez por causa do horário
- [x] Acho que a verificação da data final pode ficar direto no repositório junto com a data inicial
- [x] Acho que é bom limitarmos os caracteres do Título no cadastro da Novidade porque adicionei um com titulo grande e quebrou o layout por causa da div com o texto "Novo"

dodocinho reviewed (approved):
tentei passar o HOC `withUser` pra dentro do componente Novidade, pra evitar ficar pegando o user em toda página, mas isso quebrou a tipagem do componente e precisaria de um _refactor_ nos tipos 😔 então deixei como tava mesmo.
Acho que vale pensarmos nisso em futuras _sprints_ pra evitar a dependência de propriedades.
Além do que a Kellyn notou no review dela não tenho nada a acrescentar,  por mim ajustando essas coisas que ela listou tá aprovado.

ThaisMap committed:
fix: ajusta uso de collection nos componentes

- Ajusta Form para salvar a data inicial como o inicio do dia e a hora final como o fim do dia selecionado
- Ajusta layout da novidade para não quebrar com títulos muito longos
- Ajusta o momento da dispensa de novidades com autoDispensar
- Corrige typo para novidades em /configurações
- Ajusta nome do service GetNovidadeValida para singular
- Move o filtro de novidades pela data final para o repository

kellynvd reviewed (changes requested):
Só ficou pendente ainda o ajustes do título, fiz alguns teste mas ainda "quebra" o layout

![Screenshot from 2021-06-25 09 12 12](https://user-images.githubusercontent.com/13313889/123424713-987c6800-d597-11eb-8c20-86e5e70fc1db.png)

Aqui o "Novo" ficou desalinhado
![Screenshot from 2021-06-25 09 15 50](https://user-images.githubusercontent.com/13313889/123424729-9c0fef00-d597-11eb-9c0c-bcc82459efd0.png)

ThaisMap committed:
fix: limita caracteres no titulo da novidade

ThaisMap committed:
style: ajusta o estilo das novidades

- Ajusta layout conforme telas novas no figma

kellynvd reviewed (changes requested):
Ficou certinho agora e gostei do helperText :blush: , porém quando clico em "Nova novidade" ele começa com `undefined / 25`

![Screenshot from 2021-06-29 07 55 03](https://user-images.githubusercontent.com/13313889/123786327-f1a60d80-d8af-11eb-8392-a652795fbe1d.png)

ThaisMap committed:
fix: ajusta helperText inicial do titulo
